onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -234,12 +241,10 @@ static void SortHeterogenousDXCoreAdapterList(
   std::sort(adapter_infos.begin(), adapter_infos.end(), policy);
 }
 
-std::shared_ptr<IExecutionProviderFactory> DMLProviderFactoryCreator::Create(int device_id) {
-  return Create(device_id, /*skip_software_device_check*/ false);
-}
-
 std::shared_ptr<IExecutionProviderFactory> DMLProviderFactoryCreator::CreateFromOptions(
-  OrtDmlDeviceOptions* device_options) {
+    OrtDmlDeviceOptions* device_options,
+    bool disable_metacommands,
```

Comment:
>disable_metacommands [](http://example.com/codeflow?start=9&length=20)

Should these just be a part of OrtDmlDeviceOptions?

Reply:
`OrtDmlDeviceOptions` is not carried over until the execution provider (it's understandably discarded after the device has been created, and the device itself is passed). So it doesn't really have the same lifetime.